### ISSUE TITLE
Make widelands 4k ready

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#3392]
On new 4k (3840 x 2160 px) monitors the graphics will be too small and clicking could be very difficult. This was posted on the forum several times now:

https://wl.widelands.org/forum/topic/4466/
https://wl.widelands.org/forum/topic/4441/

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2019-02-28T08:39:46Z
 * owner: franku
 * the launchpad url was https://bugs.launchpad.net/bugs/1818037

**Comment by bunnybot** [comment]
(by gunchleoc)
Prerequisites:

https://bugs.launchpad.net/widelands/+bug/1377660
https://code.launchpad.net/~widelands-dev/widelands/font_size-lua

**Comment by treefrog00** [comment]
Not sure if this works as a solution for 4k monitors, but I posted a comment here about how to manually set the font size bigger: https://github.com/widelands/widelands/issues/2851#issuecomment-1159495847

**Comment by Joachim-Otahal** [comment]
My workaround is to automatically temporary switch the resolution:
Get [NIRCMD](https://www.nirsoft.net/utils/nircmd.html), copy it to the Widelands directory.
Create a .cmd in the widelands directory containing:
```
nircmd.exe setdisplay 2560 1440 32
ping -n 2 127.1
widelands.exe
nircmd.exe setdisplay 3840 2160 32
```
Run the .cmd, and enjoy! Of course you can set your preferred resolution in line one.  (BTW: Your gfx card should be set to let your monitor do the scaling - quite some drivers have weird defaults on this)
[I was directed here from the Widelands forum.](https://www.widelands.org/forum/topic/5738)

**Comment by Diomas** [comment]
Got the same issue on a small screen with resolution 1920x1080. Maybe we can have some option like "Scale x 2" that just makes every pixel displayed as physical 4 pixels? It would make the game playable for me.

**Comment by timhorn1** [comment]
I would appreciate it if the game itself implemented some sort of content scaling. Currently, I'm using Arch + Hyprland and setting the display resolution to 1920x1080 as a workaround, as you mentioned. However, the game looks a bit more pixelated with this workaround.

I really enjoy this game, though, and am hopeful that this feature will be implemented.

**Comment by mennucc** [comment]
here is a solution 

- install using flatpak 
 ```
 flatpak install org.widelands.Widelands
 flatpak install org.freedesktop.Platform.VulkanLayer.gamescope
```
- use this script to zoom X2

```
flatpak run  --command=gamescope \
  --env=PATH="/app/bin:/usr/bin:/usr/lib/extensions/vulkan/gamescope/bin" \
  --env=LD_LIBRARY_PATH="/usr/lib/extensions/vulkan/gamescope/lib" \
  org.widelands.Widelands   -w 1920 -h 900 -W 3840 -H 1800 \
  --filter linear   -- widelands
```
  or use this script to zoom X3

```
 flatpak run --command=gamescope \
  --env=PATH="/app/bin:/usr/bin:/usr/lib/extensions/vulkan/gamescope/bin" \
  --env=LD_LIBRARY_PATH="/usr/lib/extensions/vulkan/gamescope/lib" \
  org.widelands.Widelands   -w 1280 -h 640 -W 3840 -H 1920 \
  --filter linear   -- widelands
```

- after starting widelands, go into the options, and select the exact resolution listed by -w -h , then quit, and restart

- the above are "windowed" options, add -f and set  

> -w 1920 -h 1080 -W 3840 -H 2160

 for fullscreen mode

- use this to view all gamescope options

```
 flatpak run   --command=gamescope \
  --env=PATH="/app/bin:/usr/bin:/usr/lib/extensions/vulkan/gamescope/bin" \
  --env=LD_LIBRARY_PATH="/usr/lib/extensions/vulkan/gamescope/lib" \
  org.widelands.Widelands --  /usr/lib/extensions/vulkan/gamescope/bin/gamescope --help
```

**Comment by Fr-Dae** [comment]
in addition of UI some tribe have very low resolution png 

Certain tribes, such as the Atlanteans, have units with very low resolution; the carriers are a very good example.

There are many of them, and they clearly look ‘ugly’ compared to the rest.

This is in contrast to the Empire's sportsmen or barbarians, which exist in several resolutions.

I did an upscale test to see what would happen. 

<img width="1536" height="1024" alt="Image" src="https://github.com/user-attachments/assets/5e2e0749-ab9c-47b2-9b79-1ad148a6c027" />

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/3361#issuecomment-10691861)*
*On Wed Feb 18 09:41:42 CET 2026, [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) wrote:*

---

I want to work on UI scaling this release cycle. I don't think one cycle is long enough to solve this broad topic entirely, but certainly to improve some things.
I would appreciate a review of the refactoring PR #6937 as it's a prerequisite for the approach I have in mind.

**Comment by bunnybot** [comment]
Assigned to [<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese) [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) *( `@cb/Nordfriese` )*
